presto-main/src/main/java/com/facebook/presto/sql/relational/RawExpressionUtils.java

```diff
@@ -0,0 +1,169 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.relational;
+
+import com.facebook.presto.spi.relation.RowExpression;
+import com.facebook.presto.spi.relation.RowExpressionVisitor;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.sql.tree.Expression;
+
+import java.util.Objects;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * The following has a list of helper functions to transform between Expression and RowExpression.
+ * Ideally, users should never do type checking or down casting for future convenience of refactoring.
+ */
+public final class RawExpressionUtils
+{
+    private RawExpressionUtils() {}
+
+    /**
+     * Create a new RowExpression
+     */
+    public static RowExpression toRaw(Expression expression)
+    {
+        return new RawExpression(expression);
+    }
+
+    /**
+     * Check if the given {@param rowExpression} is a raw expression.
+     */
+    public static boolean isRaw(RowExpression rowExpression)
+    {
+        return (rowExpression instanceof RawExpression);
+    }
+
+    /**
+     * If the given {@param rowExpression} is a raw expression, apply {@param consumer}
+     */
+    public static void ifRaw(RowExpression rowExpression, Consumer<Expression> consumer)
+    {
+        if (isRaw(rowExpression)) {
+            consumer.accept(((RawExpression) rowExpression).getExpression());
+        }
+    }
+
+    /**
+     * If the given {@param rowExpression} is a raw expression, apply {@param consumer}.
+     * Otherwise, apply {@param elseConsumer} on {@param rowExpression}.
+     */
+    public static void ifRawOrElse(RowExpression rowExpression, Consumer<Expression> consumer, Consumer<RowExpression> elseConsumer)
+    {
+        if (isRaw(rowExpression)) {
+            ifRaw(rowExpression, consumer);
+        }
+        else {
+            elseConsumer.accept(rowExpression);
+        }
+    }
+
+    /**
+     * Transform the Expression inside {@param rowExpression}.
+     */
+    public static RowExpression mapExpression(RowExpression rowExpression, Function<Expression, Expression> transform)
+    {
+        checkArgument(isRaw(rowExpression));
+        return toRaw(transform.apply(((RawExpression) rowExpression).getExpression()));
+    }
+
+    /**
+     * Transform the Expression inside {@param rowExpression} if it is a RawExpression.
+     * Otherwise, apply {@param elseTransform} on {@param rowExpression}.
+     */
+    public static RowExpression mapExpressionOrElse(RowExpression rowExpression, Function<Expression, Expression> transform, Function<RowExpression, RowExpression> elseTransform)
```

Comment:
This helper and the following would make the code much harder to read and understand where they are used (You have to read this code or the comment to know what they are doing). This also made them less likely to be reused later. So I'd prefer to just write out the logic where these methods are called. Use for loop if stream interface is hard to write / understand.